python/lsst/ts/electrometer/controller.py

```diff
@@ -582,15 +580,18 @@ async def check_error(self):
     async def get_mode(self):
         """Get the mode/unit."""
         res = await self.send_command(f"{self.commands.get_mode()}", has_reply=True)
+        self.log.debug(f"Mode returns {res}")
         if res not in ['"CHAR"', '"RES"']:
             mode, unit = res.split(":")
             mode = mode.replace('"', "")
         else:
             mode = res
             mode = mode.replace('"', "")
+
         self.mode = enums.UnitMode(mode)
         await self.csc.evt_measureType.set_write(
-            mode=list(self.modes.values()).index(self.mode), force_output=True
+            mode=int([num for num, mode in self.modes.items() if self.mode == mode][0]),
```

Comment:
The code is generating an iterator filled with int values of each mode since the CSC uses int enums but only if that mode int corresponds with the current mode which leaves a list of length of one. It then returns that first index.

Reply:
That's correct. This is the most pythonic way I know to do this.